configure

```diff
@@ -7,4 +7,5 @@ if [ $# -ne 2 ]; then
 else
     echo "OSNAME   = $1" > Makefile.config
     echo "ARCHNAME = $2" >> Makefile.config
+    echo "WORDSIZE = $(getconf LONG_BIT)" >> Makefile.config
```

Comment:
I think the `WORDSIZE` should be derived from `ARCHNAME` not from the host system `ps` is compiled on.

Reply:
However, ```ARCHNAME``` only supports x86.
Additionally, ```x86``` and ```x86_64``` share almost identical code.
Not sure if we want to add another option to ```ARCHNAME```

Reply:
This is mostly target architecture specific + a `ps` requirement, so I don't think host system's configuration should influence that. 
You could be compiling `ps` on a `64bit` machine for a `32bit` target machine.

We don't need another option until we do. ;)
For starters, we could set `WORDSIZE=32` for `x86` and `WORDSIZE=64` for `x86-64` as default. 
To allow `WORDSIZE=32` for `x86-64` architecture, we could allow an optional config parameter that overrides that default, like `./configure cos x86-64 32` also limiting the range of values for that `WORDSIZE` parameter.

Reply:
It seems reasonable that if we're compiling for x86-64 to use 64 bit.  Is specifying the architecture sufficient information to figure out which bit width you want?

If so, then creating an `x86_64` directory that simply has a `ps_plat.h` which just 

```
#define WORD_SIZE 64
#include <ps_plat_common.h>
```

(similar with x86), and having `ps_plat.h` as it currently exists instead be `ps_plat_common.h` in `x86/`, and symlinked into `x86_64/`.  Sorry, I'm rushing, but does this make sense?

Is this the best solution?

Reply:
If we want another directory for ```x86_64```, I'd suggest to move this to different directory.
https://github.com/gwsystems/ps/blob/master/plat/arch/x86/ps_arch.h#L45-L51
Then, we do not need ```WORD_SIZE ``` at all, because ```PS_WORDSIZE``` is only used there.